samples/csharp_dotnetcore/CoreBot-App-Insights/Startup.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.ApplicationInsights;
+using Microsoft.Bot.Builder.Integration.ApplicationInsights.Core;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.BotBuilderSamples.Bots;
+using Microsoft.BotBuilderSamples.Dialogs;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+
+            // Create the credential provider to be used with the Bot Framework Adapter.
+            services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
+
+            // Add Application Insights services into service collection
+            services.AddApplicationInsightsTelemetry();
+
+            // Create the telemetry client.
+            services.AddSingleton<IBotTelemetryClient, BotTelemetryClient>();
+
+            // Create http midleware
+            services.AddTransient<TelemetrySaveBodyASPMiddleware>();
```

Comment:
>TelemetrySaveBodyASPMiddleware [](start = 34, length = 30)

(Low Pri) If we're going to boot in this way, may make sense to add more docs to this middleware :)  This was wrapped in the other helper before.  But like you mention, generally useful so makes sense to expose..